runner/runner.go

```diff
@@ -1514,28 +1514,39 @@ retry:
 	var responsePath string
 	if scanopts.StoreResponse || scanopts.StoreChain {
 		domainFile := strings.ReplaceAll(urlutil.TrimScheme(URL.String()), ":", ".")
-
-		// On various OS the file max file name length is 255 - https://serverfault.com/questions/9546/filename-length-limits-on-linux
-		// Truncating length at 255
-		if len(domainFile) >= maxFileNameLength {
-			// leaving last 4 bytes free to append ".txt"
-			domainFile = domainFile[:maxFileNameLength]
-		}
-
-		domainFile = strings.ReplaceAll(domainFile, "/", "[slash]") + ".txt"
+		hash := hashes.Sha256([]byte(domainFile))
+		domainFile = fmt.Sprintf("%s.txt", hash)
+		domainBaseDir := filepath.Join(scanopts.StoreResponseDirectory, URL.Host)
 		// store response
-		responsePath = filepath.Join(scanopts.StoreResponseDirectory, domainFile)
+		responsePath = filepath.Join(domainBaseDir, domainFile)
 		respRaw := resp.Raw
+		reqRaw := requestDump
 		if len(respRaw) > scanopts.MaxResponseBodySizeToSave {
 			respRaw = respRaw[:scanopts.MaxResponseBodySizeToSave]
 		}
-		writeErr := os.WriteFile(responsePath, []byte(respRaw), 0644)
+		data := append([]byte(fmt.Sprintf("[%s]", fullURL)), append([]byte("\n\n"), reqRaw...)...)
+		data = append(data, append([]byte("\n"), respRaw...)...)
+		_ = fileutil.CreateFolder(domainBaseDir)
+		writeErr := os.WriteFile(responsePath, data, 0644)
 		if writeErr != nil {
 			gologger.Error().Msgf("Could not write response at path '%s', to disk: %s", responsePath, writeErr)
 		}
+		status := strings.ReplaceAll(strings.Split(resp.RawHeaders, "\n")[0], "HTTP/1.1 ", "")
+		status = strings.ReplaceAll(status, "HTTP/2 ", "")
+		indexData := fmt.Sprintf("%s %s (%s)\n", responsePath, fullURL, status)
```

Comment:
Looks like `status` contains a trailing newline that needs to be removed before writing the data:
```
...
out\192.168.1.1\c5eb5a4cc76a5cdb16e79864b9ccd26c3553f0c396d0a21bafb7be71c1efcd8c.txt https://192.168.1.1:443 (200 OK
)
...
```

Reply:
Done